packages/split-layout/src/vaadin-split-layout-mixin.js

```diff
@@ -63,17 +63,39 @@ export const SplitLayoutMixin = (superClass) =>
 
     /** @private */
     _processChildren() {
-      [...this.children].forEach((child, i) => {
-        if (i === 0) {
-          this._primaryChild = child;
-          child.setAttribute('slot', 'primary');
-        } else if (i === 1) {
-          this._secondaryChild = child;
-          child.setAttribute('slot', 'secondary');
-        } else {
-          child.removeAttribute('slot');
-        }
-      });
+      const children = [...this.children];
+
+      children
+        .filter((child) => child.hasAttribute('slot'))
+        .forEach((child) => {
+          const slot = child.getAttribute('slot');
+          if (child.__autoSlotted) {
+            this[`_${slot}Child`] = null;
+            child.removeAttribute('slot');
+          } else {
+            this[`_${slot}Child`] = child;
+          }
+        });
+
+      children
+        .filter((child) => !child.hasAttribute('slot'))
+        .forEach((child, i) => {
+          if (this._primaryChild && this._secondaryChild) {
+            child.removeAttribute('slot');
+            return;
```

Comment:
This block seems to be not needed. All the tests pass without it, also `removeAttribute` doesn't make much sense here as the `children` in this case are filtered by `!hasAttribute(slot)`.

Reply:
You are right! That's not needed.